### PR TITLE
Improve the net crate

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16214)

### CODE REVIEW

**Comment by nox** [comment]
I added some commits that do the same thing for `Response`.

**Comment by nox** [comment]
Added a couple of commits again.

File: components/net/fetch/methods.rs [hunk 1]
```diff
@@ -196,63 +197,61 @@ pub fn main_fetch(request: &mut Request,
...
+            *origin == current_url.origin()
+        } else {
+            false
+        };
 
-            } else {
-                request.response_tainting = ResponseTainting::CorsTainting;
-                http_fetch(request, cache, true, false, false, target, done_chan, context)
+        if (same_origin && !cors_flag ) ||
+            current_url.scheme() == "data" ||
```
Comment (emilio) [inline components/net/fetch/methods.rs]: While you're at it, you could make this use `matches!`.

**Review by emilio — APPROVED** [review]
Awesome cleanup :)

**Comment by nox** [comment]
This code is probably going to change again afterwards, so I'm not doing it.

@bors-servo r=emilio

**Pull request merged**